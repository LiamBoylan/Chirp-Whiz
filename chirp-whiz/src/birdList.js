var birdList = [
    {
        name: 'Evening grosbeak',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/120746601-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67271841-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67271871-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67271851-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67271821-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67271831-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/67271881-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5140so.mp3',
        habitat: 'Forests',
        birdType: 'Grosbeak',
        scientificName: 'Coccothraustes vespertinus'
    },
    {
        name: 'American goldfinch',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/124706471-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737391-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737331-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737381-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737351-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737361-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737321-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63737341-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5290so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Finch',
        scientificName: 'Carduelis tristis'        
    },
    {
        name: 'Baltimore oriole',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/63734201-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734251-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734271-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734221-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734191-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734211-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63734241-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5070so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Oriole',
        scientificName: 'Icterus galbula'
    },
    {
        name: 'Orchard oriole',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/67361081-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361111-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361071-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361061-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361231-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67361101-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5060so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Oriole',
        scientificName: 'Icterus spurius'
    },
    {
        name: 'Common Grackle',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/67364561-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67364581-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/67364531-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/67364551-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/67364591-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/67364571-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5110so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Blackbird',
        scientificName: 'Quiscalus quiscula'
    },
    {
        name: 'Wood Duck',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65533521-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857401-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857331-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/59857361-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/59857371-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857381-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857431-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857341-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857391-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59857351-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h1440so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Duck',
        scientificName: 'Aix sponsa'
    },
    {
        name: 'Crested Cormorant',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/66027281-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027241-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027251-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027231-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027271-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027261-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66027291-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h1200so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Cormorant',
        scientificName: 'Phalacrocorax auritus'
    },
    {
        name: 'Common Loon',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/63918061-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918031-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918051-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918101-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918071-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918091-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63918081-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h0070so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Loon',
        scientificName: 'Gavia immer'
    },
    {
        name: 'Common Merganser',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/63910971-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63910941-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/63910951-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63911001-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63910981-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63910991-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63910961-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h1290so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Duck',
        scientificName: 'Mergus merganser'
    },
    {
        name: 'Osprey',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/60320581-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60320591-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60320521-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60320571-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60320531-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60320561-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3640so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Osprey',
        scientificName: 'Pandion haliaetus'
    },
    {
        name: 'Belted Kingfisher',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65764731-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/65764771-720px.jpg',
            'https://www.allaboutbirds.org/guide/assets/photo/65764761-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65764751-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65764741-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65764781-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3900so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Kingfisher',
        scientificName: 'Ceryle alcyon'
    },
    {
        name: 'Purple Martin',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/68280401-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280381-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280331-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280361-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280371-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280391-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68280341-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h6110so.mp3',
        habitat: 'Lakes and Ponds',
        birdType: 'Swallow',
        scientificName: 'Progne subis'
    },
    {
        name: 'Northern Bobwhite',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65614911-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614961-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614981-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614931-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614971-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614941-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614951-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65614921-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h2890so.mp3',
        habitat: 'Grasslands',
        birdType: 'Game Bird',
        scientificName: 'Colinus virginianus'
    },
    {
        name: 'Killdeer',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/64809651-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809631-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809641-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809671-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809691-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809661-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64809711-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h2730so.mp3',
        habitat: 'Grasslands',
        birdType: 'Plover',
        scientificName: 'Charadrius vociferus'
    },
    {
        name: 'Eastern Kingbird',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65684501-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65684471-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65684491-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65684461-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65684481-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4440so.mp3',
        habitat: 'Grasslands',
        birdType: 'Flycatcher',
        scientificName: 'Tyrannus tyrannus'
    },
    {
        name: 'Barn Swallow',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/68123101-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123061-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123031-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123021-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123071-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68123081-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h6130so.mp3',
        habitat: 'Grasslands',
        birdType: 'Swallow',
        scientificName: 'Hirundo rustica'
    },
    {
        name: 'Eastern Bluebird',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/63740061-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740051-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740021-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740031-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740071-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740081-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63740091-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h7660so.mp3',
        habitat: 'Grasslands',
        birdType: 'Thrush',
        scientificName: 'Sialia sialis'
    },
    {
        name: 'Grasshopper Sparrow',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/39344721-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/32804021-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/39347041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/39346411-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/32803981-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/32804001-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h5460so.mp3',
        habitat: 'Grasslands',
        birdType: 'Sparrow',
        scientificName: 'Ammodramus savannarum'
    },
    {
        name: 'Mourning Dove',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/60386921-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60386901-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60386931-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60386911-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60386941-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60386951-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3160so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Dove',
        scientificName: 'Zenaida macroura'
    },
    {
        name: 'Yellow-Billed Cuckoo',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/66038331-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66038311-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66038321-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66038301-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/66038341-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3870so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Cuckoo',
        scientificName: 'Coccyzus americanus'
    },
    {
        name: 'Chuck-Will`s-Widow',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/32803711-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/37178631-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/37178651-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4160so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Nightjar',
        scientificName: 'Caprimulgus carolinensis'
    },
    {
        name: 'Ruby-Throated Hummingbird',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/60395561-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395531-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395511-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395581-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395571-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395521-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60395551-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4280so.mp3',
        habitat: 'Open Woodlands',
        birdType: 'Hummingbird',
        scientificName: 'Archilochus colubris'
    },
    {
        name: 'Ruffed Grouse',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65615501-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65615471-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65615511-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65615481-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65615491-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3000so.mp3',
        habitat: 'Forests',
        birdType: 'Game Bird',
        scientificName: 'Bonasa umbellus'
    },
    {
        name: 'Barred Owl',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/60394861-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394891-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394881-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394831-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394841-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394851-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60394901-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h3680so.mp3',
        habitat: 'Forests',
        birdType: 'Owl',
        scientificName: 'Strix varia'
    },
    {
        name: 'Red-Bellied Woodpecker',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/64995071-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64995061-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64995041-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64995051-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/64995081-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4090so.mp3',
        habitat: 'Forests',
        birdType: 'Woodpecker',
        scientificName: 'Melanerpes carolinus'
    },
    {
        name: 'Eastern Wood-Pewee',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/65617351-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65617361-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65617331-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/65617341-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4610so.mp3',
        habitat: 'Forests',
        birdType: 'Flycatcher',
        scientificName: 'Contopus virens'
    },
    {
        name: 'Blue Jay',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/59859171-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59859151-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59859191-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59859211-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/59859161-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h4770so.mp3',
        habitat: 'Forests',
        birdType: 'Jay',
        scientificName: 'Cyanocitta cristata'
    },
    {
        name: 'Black-Capped Chickadee',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/60411301-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60411271-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60411291-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60411261-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60411311-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/60411281-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h7350so.mp3',
        habitat: 'Forests',
        birdType: 'Chickadee',
        scientificName: 'Parus atricapillus'
    },
    {
        name: 'Tufted Titmouse',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/63745741-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63745721-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63745771-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63745731-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63745781-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/63745761-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h7310so.mp3',
        habitat: 'Forests',
        birdType: 'Titmouse',
        scientificName: 'Parus bicolor'
    },
    {
        name: 'Red-Breasted Nuthatch',
        image: ['https://www.allaboutbirds.org/guide/assets/photo/68040881-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68040941-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68040961-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68040891-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68040921-720px.jpg', 
            'https://www.allaboutbirds.org/guide/assets/photo/68040901-720px.jpg'],
        sound: 'https://www.mbr-pwrc.usgs.gov/id/htmwav/h7280so.mp3',
        habitat: 'Forests',
        birdType: 'Nuthatch',
        scientificName: 'Sitta canadensis'
    }
];

export default birdList;