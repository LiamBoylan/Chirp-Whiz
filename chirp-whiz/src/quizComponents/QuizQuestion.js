import React from 'react';
import Image from 'react-bootstrap/Image';  // Bootstrap for general styling
import AudioButton from './AudioButton';    // For bird audio
import ProgressBar from 'react-bootstrap/ProgressBar';      // Progress bar during quiz

/*
    The question pages rendered during the quiz. Renders the bird's image and/or audio
    depending on the quiz options specified in the main quiz component before the quiz
    begins. Shows the user's progress through the quiz with the current and total question
    numbers along with a progress bar.
*/

// Renders the bird's image and/or audio for the current question.
// Props are qType for whether the question includes image, audio, or both,
// bird for the current bird and its images and sound, and imageNum for the
// current image to be displayed. This value is randomized so that a different
// image for the same bird is shown on the quiz for each question.
export function QuestionInfo(props) {
    return(
        <div>
            {props.qType.includes('image') &&
                <Image src={props.bird.image[props.imageNum]} thumbnail style={{height: '300px'}}/>
            }
            {props.qType.includes('audio') &&
                <AudioButton sound={props.bird.sound} />
            }
        </div>
    );
}

// Renders the full question along with the progress through the quiz and the question info.
// Props are the current and total question numbers along with the correct bird's info in qBird,
// the question type in qType, and the randomized image number in birdImage.
function QuizQuestion(props) {
    return (
        <div>
            <h4>Question number {props.currentQuestion} of {props.questionNum}</h4>
            <ProgressBar now={(props.currentQuestion / props.questionNum) * 100} />
            <h4>Identify this bird:</h4>
            <QuestionInfo bird={props.qBird} qType={props.qType} imageNum={props.birdImage} />
        </div>
    );
}

export default QuizQuestion;