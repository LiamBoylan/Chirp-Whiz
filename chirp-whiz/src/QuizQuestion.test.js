import React from 'react';
import QuizQuestion from './quizComponents/QuizQuestion';
import birdList from './birdList';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

const QuestionInfo = require('./quizComponents/QuizQuestion.js');

// Integration tests

it('QuizQuestion snapshot', () => {
    const container = render(<QuizQuestion
        currentQuestion={2}
        questionNum={5}
        qType={['image', 'audio']}
        qBird={birdList[0]}
      />)
    expect(container.firstChild).toMatchSnapshot();
});

test("Check if question is on the page", () => {
    const {getByText} = render(<QuizQuestion
        currentQuestion={2}
        questionNum={5}
        qType={['image', 'audio']}
        qBird={birdList[0]}
      />)
    expect(getByText("Identify this bird:")).toBeInTheDocument();
});

test("Validate the current and max question numbers", () => {
    const {getByText} = render(<QuizQuestion
        currentQuestion={3}
        questionNum={10}
        qType={['image', 'audio']}
        qBird={birdList[0]}
      />)
    expect(getByText("Question number 3 of 10")).toBeInTheDocument();
});
