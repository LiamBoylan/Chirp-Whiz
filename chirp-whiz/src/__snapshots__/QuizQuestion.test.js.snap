// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizQuestion snapshot 1`] = `undefined`;
